lib/cocoapods/installer/file_references_installer.rb

```diff
@@ -184,6 +185,54 @@ def add_file_accessors_paths_to_pods_group(file_accessor_key, group_key = nil, r
         end
       end
 
+      # Filters a list of paths down to those paths which can be added to
+      # the Xcode project. Some paths are intermediates and only their children
+      # should be added, while some paths are treated as bundles and their
+      # children should not be added directly.
+      #
+      # @param  [Array<Pathname>] paths
+      #         The paths to files or directories on disk.
+      #
+      # @return [Array<Pathname>] The paths which can be added to the Xcode project
+      #
+      def allowable_project_paths(paths)
```

Comment:
This is done as a separate pass because it needs to determine which `.lproj` directories are only added directly (without adding any of its children), and therefore needs access to the entire paths list. This was to preserve the existing behavior of non-recursive `Resources/*` globs. If we don't care/stop caring about that behavior then this filtering can be added in-line in the `paths.each` on L181 and just remove all items ending in `.lproj`. I think as-implemented is the right behavior and likely always will be.